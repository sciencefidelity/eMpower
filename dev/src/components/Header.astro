---
import Navbar from "components/Navbar.astro"
const { pathname, siteName } = Astro.props as {
  pathname: string
  siteName: string
}
---
<header class="header container flex">
  <div class="gradient-background"></div>
  <h1 class="lh-1"><a href="/" class="logo text-white">{siteName}</a></h1>
  <Navbar {pathname} />
</header>

<style lang="scss">
.header {
  margin: 0 auto 7rem;
}

.logo {
  position: relative;
  z-index: 5;
  user-select: none;
  &::after {
    position: absolute;
    top: -0.1em;
    right: -0.31em;
    width: 10%;
    font-family: var(--cursive);
    color: var(--yellow);
    content: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 112 156'%3E%3Cpath d='M11 106c-2-2-3-5-3-9v-1a25 25 0 0 1 6-12 375 375 0 0 1 20-24l-3-1A199 199 0 0 0 8 43l-4-4c-2-1-2-3-3-5l-1-4 1-6 3-3 2-1h8l3 1 3 2 2 2a20 20 0 0 0 5 4l4 2 3 3v-9a66 66 0 0 1 0-12l1-5 3-4c1-2 2-3 4-3l3-1h3l3 1 4 2 2 3 1 5v2a284 284 0 0 1-2 12v12l2-1 2-3 2-3 1-4 8-8c3-3 6-4 10-4l6 1 2 1 3 1 1 1 2 3-6 9-7 9-7 8-7 10 4 3a103 103 0 0 1 4 3l4 4 13 9c4 2 7 6 11 10l5 7 1 3 2 3-1 5-3 4c-2 2-5 3-9 3-2 0-3 0-5-2a387 387 0 0 0-16-10l-4-4a178 178 0 0 0-9-9 20 20 0 0 0-6-3c-1 0-2 1-2 3v10a109 109 0 0 0 0 11 252 252 0 0 1 3 29 55 55 0 0 1 0 7l-1 3c0 4-3 6-6 9l-5 1c-3 0-6-1-8-4l-4-8a39 39 0 0 1-3-11l1-4v-25l-1-3-2 3-2 1v1l-2 3-3 3-4 2-5-4-4-4Z' style='fill:%23fddc59'/%3E%3C/svg%3E");
  }
}

.gradient-background {
  position: absolute;
  left: 0;
  z-index: 1;
  width: 100vw;
  height: 90vh;
  background: rgb(255, 255, 255);
  background: linear-gradient(0deg, var(--white) 10px, var(--blue) 100%);
}
</style>
