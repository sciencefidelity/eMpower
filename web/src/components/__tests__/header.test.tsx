/**
 * @jest-environment jsdom
 */

import React from "react"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import Header from "../header"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Header siteTitle="eMpower" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

it("renders the site title", () => {
  const siteTitle = `eMpower`
  const { container, getByText } = render(<Header siteTitle={siteTitle} />)
  expect(getByText("eMpower")).toBeInTheDocument()
  expect(container.firstChild).toMatchInlineSnapshot(`
    <header>
      <h1>
        <a
          href="/"
        >
          eMpower
        </a>
      </h1>
    </header>
  `)
})
