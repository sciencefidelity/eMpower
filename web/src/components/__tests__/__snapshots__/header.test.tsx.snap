// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div>
    <h1>
      <a
        href="/"
      >
        Default Starter
      </a>
    </h1>
  </div>
</header>
`;
